### ISSUE TITLE
cookbooks uploaded to community with `knife cookbook site` from OSX cannot be used in kitchen.

### ISSUE THREAD

**Issue by paulczar** [chef/chef#1660]
I pushed the `logstash 0.9.1` cookbook to community using the `knife cookbook site` command from an OSX box.  When I try to run kitchen from a wrapper cookbook that requires it, the wrapper cookbook fails to run because of `pax headers`.

```
-----> Starting Kitchen (v1.2.2.dev)
-----> Converging <default-ubuntu-1204>...
       Preparing files for transfer
       Resolving cookbook dependencies with Berkshelf 3.1.4...
       Removing non-cookbook files before transfer
       Transferring files to <default-ubuntu-1204>
[2014-07-17T13:46:43+00:00] INFO: Forking chef instance to converge...       
[2014-07-17T13:46:43+00:00] WARN:        
* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *        
SSL validation of HTTPS requests is disabled. HTTPS connections are still       
encrypted, but chef is not able to detect forged replies or man in the middle       
attacks.       

To fix this issue add an entry like this to your configuration file:       

  # Verify all HTTPS connections (recommended)       
  ssl_verify_mode :verify_peer       

  # OR, Verify only connections to chef-server       
  verify_api_cert true       


       To check your SSL configuration, or troubleshoot errors, you can use the
       `knife ssl check` command like so:


         knife ssl check -c /tmp/kitchen/solo.rb


       * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * 

Starting Chef Client, version 11.12.8       
[2014-07-17T13:46:43+00:00] INFO: *** Chef 11.12.8 ***       
[2014-07-17T13:46:43+00:00] INFO: Chef-client pid: 1367       
[2014-07-17T13:46:48+00:00] INFO: Setting the run_list to ["recipe[loltest::default]"] from CLI options       
[2014-07-17T13:46:48+00:00] INFO: Run List is [recipe[loltest::default]]       
[2014-07-17T13:46:48+00:00] INFO: Run List expands to [loltest::default]       
[2014-07-17T13:46:48+00:00] INFO: Starting Chef Run for default-ubuntu-1204       
[2014-07-17T13:46:48+00:00] INFO: Running start handlers       
[2014-07-17T13:46:48+00:00] INFO: Start handlers complete.       
Compiling Cookbooks...       
[2014-07-17T13:46:48+00:00] INFO: WindowsPackage light-weight resource already initialized -- overriding!       

================================================================================       
Recipe Compile Error in /tmp/kitchen/cookbooks/logstash/providers/PaxHeader/config.rb       
================================================================================       


SyntaxError       
-----------       
/tmp/kitchen/cookbooks/logstash/providers/PaxHeader/config.rb:1: syntax error, unexpected tIDENTIFIER, expecting $end       
18 gid=1604020879       
      ^       

```

I bumped the version to `0.9.2` and reuploaded from an Ubuntu box and then it works fine.   Given that `chefdk` supports OSX this needs to be fixed so we get the same experience uploading cookbooks from either OS.

Here's a dumb test wrapper cookbook that incudes the `logstash` cookbook that I used to test this out - https://github.com/paulczar/logstash-debug-repo   the master branch depends on the broken version,  the `092` branch depends on the newer fixed version.

**Comment by Poohblah** [comment]
Are there any workarounds for this at the moment? I'm running into this problem with a community dependency that I can't simply re-upload to the Supermarket.

**Comment by Poohblah** [comment]
One workaround that I've found is to remove any directories called "PaxHeader" after installing cookbooks.  If you're using Policyfile, then you would do something like:

```
chef install
find ~/.chefdk/cache/cookbooks -type d -name PaxHeader -exec rm -r {} +
chef update
```

I'm not sure what the equivalent would be with Berks, but it should be pretty similar to the above.

I've also submitted a PR, referenced above.

One long-term solution would be to remove dependencies on Minitar, which can't handle anything beyond what's specified by POSIX.

**Comment by Poohblah** [comment]
What's the status on this? There doesn't appear to have been an official update since 2014.

**Comment by lamont-granquist** [comment]
does putting PaxHeader in a chefignore file fix this?

**Comment by Poohblah** [comment]
Unfortunately that does not fix it for me.  (I'm using Policyfile. I haven't triggered the issue with Berks.)

**Comment by specialkevin** [comment]
I also ran into this problem with the zap cookbook from the community supermarket. I would greatly appreciate a fix other than going into the cache and removing the PaxHeader files.

**Comment by lamont-granquist** [comment]
needs to be replicated on recent chef and submitted as a new issue, and should reference as to if chefignore'ing PaxHeader works correctly or not.

**Comment by Poohblah** [comment]
This has been fixed as far as I can tell.  The issue was resolved in the cookbook-omnifetch gem version 0.4.0.  I'm on Chef DK 3.1.0, which includes cookbook-omnifetch 0.8.0.
